src/plugins/saved_objects_management/public/lib/resolve_saved_objects.ts

```diff
@@ -84,50 +84,53 @@ async function importIndexPattern(
     type,
     typeMeta,
   } = doc._source;
-  const importedIndexPattern = {
+  const indexPatternSpec: IndexPatternSpec = {
     id: doc._id,
     title,
     timeFieldName,
-  } as IIndexPattern;
+  };
+  let emptyPattern: IndexPattern;
   if (type) {
-    importedIndexPattern.type = type;
+    indexPatternSpec.type = type;
   }
-  addJsonFieldToIndexPattern(importedIndexPattern, fields, 'fields', title);
-  addJsonFieldToIndexPattern(importedIndexPattern, fieldFormatMap, 'fieldFormatMap', title);
-  addJsonFieldToIndexPattern(importedIndexPattern, sourceFilters, 'sourceFilters', title);
-  addJsonFieldToIndexPattern(importedIndexPattern, typeMeta, 'typeMeta', title);
-  Object.assign(emptyPattern, importedIndexPattern);
-
-  let newId = await emptyPattern.create(overwriteAll);
-  if (!newId) {
-    // We can override and we want to prompt for confirmation
-    const isConfirmed = await openConfirm(
-      i18n.translate('savedObjectsManagement.indexPattern.confirmOverwriteLabel', {
-        values: { title },
-        defaultMessage: "Are you sure you want to overwrite '{title}'?",
-      }),
-      {
-        title: i18n.translate('savedObjectsManagement.indexPattern.confirmOverwriteTitle', {
-          defaultMessage: 'Overwrite {type}?',
-          values: { type },
+  addJsonFieldToIndexPattern(indexPatternSpec, fields, 'fields', title);
+  addJsonFieldToIndexPattern(indexPatternSpec, fieldFormatMap, 'fieldFormatMap', title);
+  addJsonFieldToIndexPattern(indexPatternSpec, sourceFilters, 'sourceFilters', title);
+  addJsonFieldToIndexPattern(indexPatternSpec, typeMeta, 'typeMeta', title);
+  try {
+    emptyPattern = await indexPatterns.createAndSave(indexPatternSpec, overwriteAll, true);
+  } catch (err) {
+    if (err instanceof DuplicateIndexPatternError) {
```

Comment:
optional: does it make sense to provide an error helper instead to hide an implementation details?
```js
dataPluginErrors.indexPatterns.isDuplicate(err) ? 
```

Reply:
I think this is a code style question. In this case `DuplicateIndexPatternError` is being treated as a public interface. The code is trivial. I think code consistency is more important than which path we choose.